src/haddock/libs/libio.py

```diff
@@ -501,3 +502,28 @@ def file_exists(
 
     # don't change to f-strings, .format has a purpose
     raise exception(emsg.format(str(path)))
+
+
+def get_perm(fname):
+    """Get permissions of file."""
+    # https://stackoverflow.com/questions/6874970
+    return stat.S_IMODE(os.lstat(fname)[stat.ST_MODE])
+
+
+def make_writeable_recursive(path):
```

Comment:
Isn't there an os function to do it already? Just curious. (just like `chmod -R` as the shell level)

Reply:
I could have called the shell, but I want to keep things inside python as much as possible so that python handles the different OSes properly.